src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -297,7 +300,7 @@ public ReadOnlySpan<T> Peek(int maxCount)
             }
 
             // Not enough contiguous Ts, allocate and copy what we can get
-            return PeekSlow(new T[maxCount]);
+            return PeekSlow(_arrayPool.Rent(maxCount));
```

Comment:
How do we plan to return the array back to the pool?

Reply:
We don't- see my other comments.